packages/crawler/src/crawler/crawler-engine.ts

```diff
@@ -16,38 +16,40 @@ export type ApifyMainFunc = (userFunc: Apify.UserFunc) => void;
 
 export class CrawlerEngine {
     public constructor(
+        private readonly logger: Logger,
         private readonly pageProcessorFactory: PageProcessorFactory = new PageProcessorFactory(),
         private readonly crawlerFactory: CrawlerFactory = new CrawlerFactory(),
         private readonly resourceCreator: ResourceCreator = new ApifyResourceCreator(),
         private readonly runApify: ApifyMainFunc = Apify.main,
     ) {}
 
     public async start(crawlerRunOptions: CrawlerRunOptions): Promise<void> {
-        if (!isNil(crawlerRunOptions.localOutputDir)) {
-            this.setLocalOutputDir(crawlerRunOptions.localOutputDir);
-        }
+        this.setEnvVars(crawlerRunOptions.localOutputDir);
 
         const requestList = await this.resourceCreator.createRequestList(crawlerRunOptions.existingUrls);
         const requestQueue = await this.resourceCreator.createRequestQueue(crawlerRunOptions.baseUrl);
 
-        const pageProcessor = this.pageProcessorFactory.createPageProcessor({
-            requestQueue,
-            crawlerRunOptions,
-        });
+        const pageProcessor = this.pageProcessorFactory.createPageProcessor(
+            {
+                requestQueue,
+                crawlerRunOptions,
+            },
+            this.logger,
+        );
 
         this.runApify(async () => {
             const crawler = this.crawlerFactory.createPuppeteerCrawler({
                 requestList,
                 requestQueue,
-                handlePageFunction: pageProcessor.pageProcessor,
+                handlePageFunction: pageProcessor.pageHandler,
                 gotoFunction: pageProcessor.gotoFunction,
                 handleFailedRequestFunction: pageProcessor.pageErrorProcessor,
             });
             await crawler.run();
         });
     }
 
-    private setLocalOutputDir(outputDir: string): void {
-        setApifyEnvVars({ APIFY_LOCAL_STORAGE_DIR: outputDir });
+    private setEnvVars(localOutputDir?: string): void {
```

Comment:
can you move this to crawlerConfiguration and rename it to "setOutputDirectory"?
if you think it's out of scope, we can do it in another PR